crates/services/executor/src/executor.rs

```diff
@@ -648,18 +667,115 @@ where
+                                );
+                            }
+                        }
                     }
                 }
             }
         }
 
         execution_data.event_inbox_root = root_calculator.root().into();
 
+        Ok(checked_forced_txs)
+    }
+
+    /// Parse forced transaction payloads and perform basic checks
+    fn validate_forced_tx(
+        relayed_tx: RelayedTransaction,
+        header: &PartialBlockHeader,
+        consensus_params: &ConsensusParameters,
+    ) -> Result<CheckedTransaction, ForcedTransactionFailure> {
+        let parsed_tx = Self::parse_tx_bytes(&relayed_tx)?;
+        let checked_tx =
+            Self::get_checked_tx(parsed_tx, *header.height(), consensus_params)?;
+        Self::tx_is_valid_variant(&checked_tx)?;
+        Self::relayed_tx_claimed_enough_max_gas(&checked_tx, &relayed_tx)?;
+        let predicate_checked_tx =
+            Self::predicates_on_tx_are_valid(checked_tx, consensus_params)?;
+        Ok(CheckedTransaction::from(predicate_checked_tx))
+    }
+
+    fn parse_tx_bytes(
+        relayed_transaction: &RelayedTransaction,
+    ) -> Result<Transaction, ForcedTransactionFailure> {
+        let tx_bytes = relayed_transaction.serialized_transaction();
+        let tx = Transaction::from_bytes(tx_bytes)
+            .map_err(|_| ForcedTransactionFailure::CodecError)?;
+        Ok(tx)
+    }
+
+    fn get_checked_tx(
+        tx: Transaction,
+        height: BlockHeight,
+        consensus_params: &ConsensusParameters,
+    ) -> Result<Checked<Transaction>, ForcedTransactionFailure> {
+        let checked_tx = tx
+            .into_checked(height, consensus_params)
+            .map_err(ForcedTransactionFailure::CheckError)?;
+        Ok(checked_tx)
+    }
+
+    fn tx_is_valid_variant(
+        tx: &Checked<Transaction>,
+    ) -> Result<(), ForcedTransactionFailure> {
+        match tx.transaction() {
+            Transaction::Mint(_) => Err(ForcedTransactionFailure::InvalidTransactionType),
+            Transaction::Script(_) | Transaction::Create(_) => Ok(()),
+        }
+    }
+
+    fn relayed_tx_claimed_enough_max_gas(
+        tx: &Checked<Transaction>,
+        relayed_tx: &RelayedTransaction,
+    ) -> Result<(), ForcedTransactionFailure> {
+        let claimed_max_gas = relayed_tx.max_gas();
+        let actual_max_gas = match tx.metadata() {
+            CheckedMetadata::Script(script_metadata) => script_metadata.max_gas,
+            CheckedMetadata::Create(create_metadata) => create_metadata.max_gas,
+            CheckedMetadata::Mint(_) => {
+                return Err(ForcedTransactionFailure::InvalidTransactionType);
+            }
+        };
+        if actual_max_gas > claimed_max_gas {
+            return Err(ForcedTransactionFailure::InsufficientMaxGas {
+                claimed_max_gas,
+                actual_max_gas,
+            });
+        }
         Ok(())
     }
 
+    fn predicates_on_tx_are_valid(
```

Comment:
Hmm, `into_checked` perfromces all checks inside. so we can remove `check_predicate`

<img width="714" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/eeb411ad-142e-495b-a0c4-06efdc4ead32">